extension/asapauthextension/go.mod

```
@@ -0,0 +1,46 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/extension/asapauth
```

This probably needs to have the `extension` suffix, which is likely the reason the lint check is failing.

```suggestion
module github.com/open-telemetry/opentelemetry-collector-contrib/extension/asapauthextension
```